miner/worker.go

```diff
@@ -288,6 +295,29 @@ func (miner *Miner) commitTransaction(env *environment, tx *types.Transaction) e
 	if tx.Type() == types.BlobTxType {
 		return miner.commitBlobTransaction(env, tx)
 	}
+
+	// If a conditional is set, check prior to applying
+	if conditional := tx.Conditional(); conditional != nil {
+		now, cost := time.Now(), conditional.Cost()
+		if !miner.conditionalLimiter.AllowN(now, cost) {
+			return fmt.Errorf("exceeded rate limit: cost %d, tokens %f: %w", cost, miner.conditionalLimiter.Tokens(), errTxConditionalRateLimited)
+		}
+
+		// ditch the reservation as we've checked that we're allowed `cost` units by the limiter
+		miner.conditionalLimiter.ReserveN(now, cost)
+		txConditionalMinedTimer.UpdateSince(conditional.SubmissionTime)
+
+		// check the conditional
+		if err := env.header.CheckTransactionConditional(conditional); err != nil {
+			conditional.Rejected.Store(true)
+			return fmt.Errorf("failed header check: %s: %w", err, errTxConditionalInvalid)
+		}
+		if err := env.state.CheckTransactionConditional(conditional); err != nil {
+			conditional.Rejected.Store(true)
```

Comment:
This atomic is a good idea, although I am not sure if the txs are copied when being pulled into the miner from the mempool, meaning that this may not make it back to the mempool, do you know if this is the case?

Reply:
Ok i see your comment here: https://github.com/ethereum-optimism/op-geth/pull/330#discussion_r1706194211

Disregard this